### PR TITLE
Improve recovering state from an attribute end position.

### PR DESCRIPTION
**Pull request by KirillOsenkov**

### CODE REVIEW

File: Core/Parser/XmlTagState.cs [hunk 1]
```diff
@@ -196,7 +196,7 @@ public override XmlParserContext TryRecreateState (XObject xobject, int position
 						newEl.Attributes.AddAttribute ((XAttribute)att.ShallowCopy ());
 						continue;
 					}
-					if (att.Span.End > position) {
+					if (att.Span.End >= position) {
```
Comment (mhutch) [inline Core/Parser/XmlTagState.cs:199]:
```suggestion
					else {
```

**Review by mhutch — COMMENTED** [review]
This superficially looks good to me but I'm not sure I fully understand the code myself any more. I'm gonna try to write a test to cover this functionality before merging.

**Comment by mhutch** [comment]
Added test in #44

**Comment by KirillOsenkov** [comment]
I think you've fixed the recovery part so all that remains is the FindAttribute fix. I hope I got it right?

**Comment by mhutch** [comment]
Sorry for the delay. I took a look at why this is failing the tests. It's breaking spine recovery - when it searches for an xobject to resume from, it finds one that's already complete and can't actually be resumed.

I think the underlying problem is that this changes the semantics. `FindAtOffset` uses `Contains` semantics for all object types. Maybe we could make a `FindAtOffsetOuter` variant of the method that uses `ContainsOuter` semantics, if you have a use case that needs that?

**Comment by mhutch** [comment]
That said, Find would become ambiguous with outer semantics e.g. the position between adjacent tags `<a><b>`.

**Comment by KirillOsenkov** [comment]
I should have provided a use case where I needed this for. Of course now I don't remember so I'll need to dig in and figure it out. I think it was to determine whether we're in an attribute context when typing an attribute (for brace completion??). It very well may be that there's a better solution for what I wanted. Let me get back to you on this.

**Comment by KirillOsenkov** [comment]
I tried without this change and everything seems fine. I'll reopen if I remember what it was for.

**Pull request closed without merging**